src/main/java/onboarding/Problem6.java

```diff
@@ -1,10 +1,185 @@
 package onboarding;
 
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 public class Problem6 {
+
     public static List<String> solution(List<List<String>> forms) {
-        List<String> answer = List.of("answer");
-        return answer;
+        return new Solver6(forms).solve();
+    }
+}
+
+final class Solver6 {
+
+    /**
+     * 크루는 1명 이상 10,000명 이하이다.
+     */
+    private final List<List<String>> forms;
+
+    /**
+     * 중복 사용자의 이메일
+     */
+    private List<String> duplicateUsersEmails = new ArrayList<>();
+
+    /**
+     * 두 글자에 해당하는 이름을 저장한다.
+     */
+    private final Set<String> earlyNicknameParts = new HashSet<>();
+    private final Set<String> lateNicknameParts = new HashSet<>();
+
+    private final FormValidator validator = new FormValidator();
+
+    /*
+        생성자에서 Validation을 수행한다.
+        - 신청할 수 있는 이메일은 `email.com` 도메인으로만 제한한다.
+        - 닉네임은 한글만 가능하고 전체 길이는 1자 이상 20자 미만이다.
+     */
+    public Solver6(final List<List<String>> forms) {
+        this.forms =
+            forms
+                .stream()
+                .filter(form -> {
+                    String email = form.get(0);
+                    String nickname = form.get(1);
+                    return valid(email, nickname);
```

Comment:
```suggestion
                    .filter(form -> valid(form.get(0), form.get(1))                       )
                        .collect(Collectors.toList());
```
이렇게 바꿀 수 있겠네요! 그리고 `get(0), get(1)` 이 부분의 숫자를 상수로 바꾸는 것도 좋아보입니다.!

Reply:
0을 EMAIL_INDEX, 1을 NICKNAME_INDEX와 같은 네이밍이 적절해 보이네요.

뿐만 아니라 전체적으로 매직 넘버를 상수로 선언해서 의도를 들어내는 것이 좋을 것 같아요!